modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```
@@ -0,0 +1,366 @@
+                case MINUTES:
+                    retentionPeriods.put(duration, mins);
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, hours);
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, days);
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(duration, months);
+                    break;
+                case YEARS:
+                    retentionPeriods.put(duration, years);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        if (annotationTypes.keySet().contains(SiddhiConstants.NAMESPACE_PURGE)) {
+            if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                    getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE))) {
+                String enable = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE);
+                if (!(enable.equals("true") || enable.equals("false"))) {
+                    throw new SiddhiAppCreationException("Undefined value for enable: " + enable + "." +
+                            " Please use true or false");
+                } else {
+                    purgingEnabled = Boolean.parseBoolean(enable);
+                }
+            }
+            if (purgingEnabled) {
+                // If interval is defined, default value of 15 min will be replaced by user input value
+                if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.NAMESPACE_INTERVAL))) {
+                    String interval = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
```

What if the interval is not specified? We should throw an error?

---

As we discussed earlier, if the interval is not defined we will use a default interval to scheduled the purging task. so interval is optional.